support-models/src/main/scala/com/gu/support/workers/GiftRecipient.scala

```diff
@@ -3,55 +3,18 @@ package com.gu.support.workers
 import com.gu.i18n.Title
 import com.gu.support.encoding.DiscriminatedType
 import com.gu.support.encoding.CustomCodecs._
-import com.gu.support.workers.GiftRecipient.{DigitalSubscriptionGiftRecipient, WeeklyGiftRecipient}
+
 import io.circe._
 import org.joda.time.LocalDate
 import com.gu.support.encoding
-
-sealed trait GiftRecipient {
-  val firstName: String
-  val lastName: String
-  def asDigitalSubscriptionGiftRecipient: Option[DigitalSubscriptionGiftRecipient] =
-    this match { case a: DigitalSubscriptionGiftRecipient => Some(a); case _ => None }
-  def asWeekly: Option[WeeklyGiftRecipient] = this match { case a: WeeklyGiftRecipient => Some(a); case _ => None }
-}
+import io.circe.generic.semiauto.deriveCodec
+
+case class GiftRecipient(
+    title: Option[Title],
+    firstName: String,
+    lastName: String,
+    email: Option[String],
+)
 object GiftRecipient {
-
-  case class WeeklyGiftRecipient(
-      title: Option[Title],
-      firstName: String,
-      lastName: String,
-      email: Option[String],
-  ) extends GiftRecipient
-
-  case class DigitalSubscriptionGiftRecipient(
-      firstName: String,
-      lastName: String,
-      email: String,
-      message: Option[String],
-      deliveryDate: LocalDate,
-  ) extends GiftRecipient
-
-  val discriminatedType = new DiscriminatedType[GiftRecipient]("giftRecipientType")
-  implicit val weeklyCodec: discriminatedType.VariantCodec[WeeklyGiftRecipient] =
-    discriminatedType.variant[WeeklyGiftRecipient]("Weekly")
```

Comment:
just thinking it through
if we deploy this while a GW gift execution is processing, then the newly deployed lambdas would not check the disciminator any more so it would deserialise successfully.
And noone could be checking out a DS gift, so that wouldn't be an issue.